blocks/recentreports/recentreports.js

```diff
@@ -0,0 +1,253 @@
+
+    const reportBox = createReportBox(
+      item.COM_NAME,
+      formatPriceInRupees(item.TARGET_PRICE),
+      item.RATING,
+      formatDate(item.REP_RELEASE_DTM),
+      item.REPORT_PDF_LINK,
+      buttontitle,
+    );
+    slide.innerHTML = reportBox.innerHTML;
+    carouselItems.append(slide);
+
+    slideCount += 1;
+  });
+}
+
+async function loadCarousel(block, carouselItems) {
+  const carouselBlock = buildBlock('carousel', '');
+  carouselBlock.style.visibility = 'hidden';
+  carouselBlock.dataset.visibleSlides = block.dataset.visibleSlides || '';
+  carouselBlock.dataset.autoScroll = block.dataset.autoScroll || '';
+  carouselBlock.dataset.autoScrollDelay = block.dataset.autoScrollDelay || '';
+  carouselBlock.innerHTML = '';
+  block.classList.forEach((className) => {
+    carouselBlock.classList.add(className);
+  });
+
+  Array.from(carouselItems.children).forEach((carouselItemElement) => {
+    const divElement = document.createElement('div');
+    Array.from(carouselItemElement.children).forEach((child) => {
+      divElement.appendChild(child.cloneNode(true));
+    });
+    carouselBlock.appendChild(divElement);
+  });
+
+  const carouselBlockParent = document.createElement('div');
+  carouselBlockParent.classList.add('carousel-wrapper');
+  carouselBlockParent.appendChild(carouselBlock);
+  block.insertBefore(carouselBlockParent, block.firstChild.nextSibling);
+  decorateBlock(carouselBlock);
+
+  loadBlock(carouselBlock).then((recentreportblock) => {
+    let totalSlidesDisplayed = 0;
+    if (carouselItems.children.length >= carouselBlock.dataset.visibleSlides) {
+      totalSlidesDisplayed = carouselBlock.dataset.visibleSlides;
+      carouselBlock.style.setProperty('--displayed-slides', carouselBlock.dataset.visibleSlides);
+    } else {
+      totalSlidesDisplayed = carouselItems.children.length;
+      carouselBlock.style.setProperty('--displayed-slides', carouselItems.children.length);
+    }
+
+    if (totalSlidesDisplayed > 1) {
+      let maxWidth = 0;
+      carouselBlock.querySelectorAll('.carousel-slide .box').forEach((slide) => {
+        if (maxWidth < slide.offsetWidth) {
+          maxWidth = slide.offsetWidth;
+        }
+      });
+
+      const margin = (carouselBlock.parentElement.offsetWidth
+        - (totalSlidesDisplayed * maxWidth)) / 2;
+      carouselBlock.querySelector('.carousel-slides-container').style
+        .setProperty('margin-inline', `${margin.toString()}px`);
+    }
+    recentreportblock.style.visibility = 'visible';
+  });
+}
+
+function handleTitleConfig(titleElement, container) {
+  const titleText = titleElement.textContent.trim();
+  const title = document.createElement('h2');
+  title.textContent = titleText;
+
+  const titleWrapper = document.createElement('div');
+  titleWrapper.classList.add('title-wrapper');
+  titleWrapper.appendChild(title);
+
+  container.insertBefore(titleWrapper, container.firstChild);
+}
+
+function addDiscoverLink(discoverMoreDiv, block) {
```

Comment:
can we leverage [decorateDiscoverMore ](https://github.com/aemsites/icicidirect/blob/main/scripts/scripts.js#L102)?

Reply:
@shrotia  This is discover more link and has different styling than discover more button